libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -59,14 +59,18 @@ func WriteJSONKeys(event *beat.Event, keys map[string]interface{}, expandKeys, o
 		case "@timestamp":
 			vstr, ok := v.(string)
 			if !ok {
-				event.SetErrorWithOption(createJSONError("@timestamp not overwritten (not string)"), addErrKey)
+				event.SetErrorWithOption("@timestamp not overwritten (not string)", addErrKey)
 				continue
 			}
 
 			// @timestamp must be of format RFC3339 or ISO8601
 			ts, err := parseTimestamp(vstr)
 			if err != nil {
-				event.SetErrorWithOption(createJSONError(fmt.Sprintf("@timestamp not overwritten (parse error on %s)", vstr)), addErrKey)
+				var builder strings.Builder
```

Comment:
what was wrong with the previous implementation (calling `SetErrorWithOption`)?
This code is duplicated with lines 105-109, please extract to a helper function (or use the previous code)

Reply:
This is a good example of an early optimization. As it turns out using this vs fmt does not show any difference in the performance of the function.